src/wms/src/main/java/org/geoserver/wms/GetMap.java

```diff
@@ -988,4 +989,19 @@ protected GetMapOutputFormat getDelegate(final String outputFormat) throws Servi
         }
         return producer;
     }
+
+    private boolean checkWMSLayerMinMaxScale(WMSLayerInfo wmsLayerInfo, double mapScale) {
+        //
+        // if none configured
+        if (wmsLayerInfo.getMinScale() == null && wmsLayerInfo.getMaxScale() == null) return true;
+        if (wmsLayerInfo.getMinScale() != null)
```

Comment:
You can use ``&& mapScale < wmsLayerInfo.getMinScale()`` in the first if, there is short-circuit evaluation

Reply:
is that Null safe?